### PR TITLE
#963 : Introduce a dedicated Media Gallery page

### PR DESCRIPTION
**Pull request by konarshankar07**
### Description (*)

This PR will introduce the new standalone media gallery
### Fixed Issues (if relevant)

1. magento/adobe-stock-integration#963: Introduce a dedicated Media Gallery page

### Manual testing scenarios (*)

1. Login to Admin panel
2. Go to Content -> Elements -> Media -> Media Gallery

### CODE REVIEW

File: MediaGalleryUi/etc/acl.xml [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+ * See COPYING.txt for license details.
+ */
+-->
+<config xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="urn:magento:framework:Acl/etc/acl.xsd">
+    <acl>
+        <resources>
+            <resource id="Magento_Backend::admin">
+                <resource id="Magento_Backend::content">
+                    <resource id="Magento_MediaGalleryUi::media" title="Media" translate="title" sortOrder="15" >
+                        <resource id="Magento_MediaGalleryUi::media_gallery" title="Media Gallery" translate="title" sortOrder="10"/>
```
Comment (sivaschenko) [inline MediaGalleryUi/etc/acl.xml]: Let's use the existing `Magento_Cms::media_gallery` resource

File: MediaGalleryUi/view/adminhtml/ui_component/standalone_media_gallery_listing.xml [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+        </deps>
+    </settings>
+    <dataSource name="media_gallery_listing_data_source" component="Magento_Ui/js/grid/provider">
+        <settings>
+            <storageConfig>
+                <param name="indexField" xsi:type="string">id</param>
+            </storageConfig>
+            <updateUrl path="mui/index/render"/>
+        </settings>
+        <aclResource>Magento_MediaGalleryUi::media_gallery</aclResource>
```
Comment (sivaschenko) [inline MediaGalleryUi/view/adminhtml/ui_component/standalone_media_gallery_listing.xml]:
```suggestion
        <aclResource>Magento_Cms::media_gallery</aclResource>
```

**Review by sivaschenko — COMMENTED** [review]
Hi @konarshankar07 thanks for working on this task, please see my comments regarding the committed changes

**Comment by konarshankar07** [comment]
Hello @sivaschenko ...
As I checked further, I think we can't reuse `Magento_Cms::media_gallery` resource because we have added new group `Media`. Let me know your thoughts
image
Thanks

**Comment by sivaschenko** [comment]
@konarshankar07 the idea is to keep the access to Media gallery for admins that currently have access to it. Can't we use the existing ACL resource with the new menu items/pages ?

**Comment by konarshankar07** [comment]
Yes, we can use the existing one but suppose if user want to give access to the standalone media gallery for one user and disable for another one so in that case I think its difficult to manage

**Comment by sivaschenko** [comment]
@konarshankar07 I don't think there is a reasonable case to give access to standalone media gallery and deny access to integrated media gallery (and vice-versa)

**Comment by konarshankar07** [comment]
Okay, I will reuse the existing resource
Thanks

**Comment by konarshankar07** [comment]
Hello @sivaschenko ...
Do we also need to change the ACL for https://github.com/magento/adobe-stock-integration/blob/2e6aa43d9601a8dd0416cd4ee72208f181a4d6c3/MediaGalleryUi/Controller/Adminhtml/Directories/GetTree.php#L28 ?
So I don't think we need acl.xml file in `Magento_MediaGalleryUi` 
Correct?
Sorry for pestering you

**Comment by sivaschenko** [comment]
Yes, that's correct, thanks @konarshankar07

File: AdobeStockImageAdminUi/view/adminhtml/web/js/panel.js [hunk 3]
```diff
@@ -36,7 +36,13 @@ define([
...
             }).on('openModal', function () {
                 this.masonry().setLayoutStylesWhenLoaded();
-            }.bind(this)).applyBindings();
+            }.bind(this));
+
+            try {
+                $(this.containerId).applyBindings();
+            } catch (e) {
+                //already bounded
+            }
```
Comment (konarshankar07) [inline AdobeStockImageAdminUi/view/adminhtml/web/js/panel.js:45]: @sivaschenko ... I thought that this issue is from Magento Core issue

**Review by sivaschenko — APPROVED** [review]

**Pull request merged**